packages/datadog-plugin-hapi/src/index.js

```diff
@@ -49,9 +49,42 @@ function createWrapDispatch (tracer, config) {
 }
 
 module.exports = [
+  {
+    name: '@hapi/hapi',
+    versions: ['>=18.2'],
+    file: 'lib/request.js',
+    patch (Request, tracer, config) {
+      this.wrap(Request, 'generate', createWrapGenerate(tracer, config))
+    },
+    unpatch (Request) {
+      this.unwrap(Request, 'generate')
+    }
+  },
+  {
+    name: '@hapi/hapi',
+    versions: ['>=17.9 <18'],
+    file: 'lib/request.js',
+    patch (Request, tracer, config) {
+      this.wrap(Request, 'generate', createWrapGenerate(tracer, config))
+    },
+    unpatch (Request) {
+      this.unwrap(Request, 'generate')
+    }
+  },
+  {
+    name: 'hapi',
+    versions: ['18 - 18.1'],
+    file: 'lib/request.js',
+    patch (Request, tracer, config) {
+      this.wrap(Request, 'generate', createWrapGenerate(tracer, config))
+    },
+    unpatch (Request) {
+      this.unwrap(Request, 'generate')
+    }
+  },
   {
     name: 'hapi',
-    versions: ['>=17.1'],
+    versions: ['17.1 - 17.8'],
```

Comment:
This change is not needed since newer versions are not published to `hapi` anymore.